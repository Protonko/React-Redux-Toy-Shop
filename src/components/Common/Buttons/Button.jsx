import React from 'react';
import PropTypes from 'prop-types';

export const Button = ({title, onClick, classNames = 'button'}) => {
  return (
    <button className={classNames} onClick={onClick}>
      {title}
    </button>
  );
}

Button.propTypes = {
  title: PropTypes.string,
  onClick: PropTypes.func,
};
