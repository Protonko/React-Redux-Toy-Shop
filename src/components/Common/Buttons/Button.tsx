// types
import {TClassName, TTypesButton, TAttributesButton} from 'models/types';
import {TYPES_BUTTON} from 'models/enums';

import React, {FC} from 'react';
import classNames from 'classnames';

export interface IProps {
  title: string,
  onClick: (...args: any) => void,
  isRemoved?: boolean,
  disabled?: boolean,
  type?: TTypesButton,
  classNamesAdditional?: TClassName,
  attrs?: TAttributesButton,
}

export const Button: FC<IProps> = ({
  title,
  onClick,
  isRemoved,
  disabled,
  type = TYPES_BUTTON.BUTTON,
  classNamesAdditional = [],
  ...attrs
}) => {
  const classNamesButton = classNames(
    'button',
    {'button--disabled': disabled},
    {'button--red': isRemoved},
    ...classNamesAdditional,
  );

  return (
    <button
      className={classNamesButton}
      onClick={onClick}
      type={type}
      {...attrs}
    >
      {title}
    </button>
  );
}
