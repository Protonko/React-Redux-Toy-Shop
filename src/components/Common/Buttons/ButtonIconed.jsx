import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

export const ButtonIconed = ({
  children,
  onClick,
  disabled,
  classNamesAdditional = [],
  ...attrs
}) => {
  const classNamesButton = classNames(
    'button',
    'button--iconed',
    {'button--disabled': disabled},
    ...classNamesAdditional,
  );

  return (
    <button className={classNamesButton} onClick={onClick} {...attrs}>
      {children}
    </button>
  );
}

ButtonIconed.propTypes = {
  onClick: PropTypes.func,
  disabled: PropTypes.bool,
  classNamesAdditional: PropTypes.array,
};
