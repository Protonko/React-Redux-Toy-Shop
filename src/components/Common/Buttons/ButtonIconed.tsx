// types
import {TClassName, TTypesButton, TAttributesButton} from 'models/types';
import {TYPES_BUTTON} from 'models/enums';

import React, {FC, ReactNode} from 'react';
import classNames from 'classnames';

export interface IProps {
  children: ReactNode,
  onClick: (...args: any) => void,
  disabled?: boolean,
  type?: TTypesButton,
  classNamesAdditional?: TClassName,
  attrs?: TAttributesButton,
}

export const ButtonIconed: FC<IProps> = ({
  children,
  onClick,
  disabled,
  classNamesAdditional = [],
  type= TYPES_BUTTON.BUTTON,
  ...attrs
}) => {
  const classNamesButton = classNames(
    'button',
    'button--iconed',
    {'button--disabled': disabled},
    ...classNamesAdditional,
  );

  return (
    <button
      className={classNamesButton}
      onClick={onClick}
      type={type}
      {...attrs}
    >
      {children}
    </button>
  );
}
