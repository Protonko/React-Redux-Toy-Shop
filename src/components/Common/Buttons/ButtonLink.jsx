import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

export const ButtonLink = ({
  text,
  disabled,
  isRemoved,
  onClick,
  classNamesAdditional = []
}) => {
  const classNamesButton = classNames(
    'button',
    'button--linked',
    {'button--colored-red': isRemoved},
    {'button--disabled': disabled},
    ...classNamesAdditional,
  );

  return (
    <button className={classNamesButton} onClick={onClick}>
      {text}
    </button>
  )
}

ButtonLink.propTypes = {
  text: PropTypes.string,
  disabled: PropTypes.bool,
  isRemoved: PropTypes.bool,
  onClick: PropTypes.func,
  classNamesAdditional: PropTypes.array,
};
