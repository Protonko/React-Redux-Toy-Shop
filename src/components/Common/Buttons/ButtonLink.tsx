// types
import {TAttributesButton, TClassName, TTypesButton} from 'models/types';
import {TYPES_BUTTON} from 'models/enums';

import React, {FC} from 'react';
import classNames from 'classnames';

export interface IProps {
  text: string,
  disabled?: boolean,
  isRemoved?: boolean,
  onClick: (...args: any) => void,
  type?: TTypesButton,
  classNamesAdditional?: TClassName,
  attrs?: TAttributesButton,
}

export const ButtonLink: FC<IProps> = ({
  text,
  disabled,
  isRemoved,
  onClick,
  type = TYPES_BUTTON.BUTTON,
  classNamesAdditional = [],
  ...attrs
}) => {
  const classNamesButton = classNames(
    'button',
    'button--linked',
    {'button--colored-red': isRemoved},
    {'button--disabled': disabled},
    ...classNamesAdditional,
  );

  return (
    <button
      className={classNamesButton}
      onClick={onClick}
      type={type}
      {...attrs}
    >
      {text}
    </button>
  )
}
