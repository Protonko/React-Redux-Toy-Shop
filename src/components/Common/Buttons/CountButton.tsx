// types
import {TAttributesButton, TTypesButton} from 'models/types';
import {TYPES_BUTTON} from 'models/enums';

import React, {FC} from 'react';
import classNames from 'classnames';

export interface IProps {
  isMinus?: boolean,
  onClick: () => void,
  type?: TTypesButton,
  attrs?: TAttributesButton,
}

export const CountButton: FC<IProps> = ({
  isMinus,
  onClick,
  type = TYPES_BUTTON.BUTTON,
  ...attrs
}) => {
  const classNamesButton: string = classNames(
      'button',
      'button--count',
      {'button--count-minus': isMinus}
  );

  return (
      <button
        className={classNamesButton}
        onClick={onClick}
        data-symbol={isMinus ? '-' : '+'}
        type={type}
        {...attrs}
      />
  );
};
