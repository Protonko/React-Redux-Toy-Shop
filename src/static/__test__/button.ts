export const ELEMENT_SELECTORS_BUTTON = {
  button: 'button',
  testButton: 'test-button',
  minusButton: 'button--count-minus',
}
