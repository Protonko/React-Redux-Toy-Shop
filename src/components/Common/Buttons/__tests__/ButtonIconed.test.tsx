import React from 'react';
import {shallow} from 'enzyme';
import {ELEMENT_SELECTORS_BUTTON} from 'static/__test__/button';
import {ButtonIconed, IProps} from 'components/Common/Buttons/ButtonIconed';

const shallowCardComponent = (props: IProps) => (
  shallow(<ButtonIconed {...props} />)
)

describe('ButtonIconed', () => {
  let props: IProps;
  let fn: jest.Mock<void, []>;

  beforeEach(() => {
    fn = jest.fn(() => {});
    props = {
      children: <div />,
      onClick: fn,
    }
  });

  it('Match snapshot', () => {
    const component = shallowCardComponent(props);

    expect(component).toMatchSnapshot();
  });

  it('Should add className "test-button"', () => {
    props = {...props, classNamesAdditional: [ELEMENT_SELECTORS_BUTTON.testButton]}
    const buttonIconedComponent = shallowCardComponent(props);
    const classNameButton = buttonIconedComponent
      .find(ELEMENT_SELECTORS_BUTTON.button)
      .hasClass(ELEMENT_SELECTORS_BUTTON.testButton);

    expect(classNameButton).toEqual(true);
  });
});
