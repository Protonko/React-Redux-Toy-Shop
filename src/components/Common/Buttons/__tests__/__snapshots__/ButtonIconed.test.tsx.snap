// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonIconed Match snapshot 1`] = `
<button
  className="button button--iconed"
  onClick={[MockFunction]}
  type="button"
>
  <div />
</button>
`;
